src/patternfly/components/DataList/data-list.scss

```diff
@@ -298,13 +298,15 @@
   margin-left: var(--pf-c-data-list__toggle--MarginLeft);
 
   // rotate icon
-  .pf-c-button > * {
+  .pf-c-button .pf-c-data-list__toggle-icon {
```

Comment:
You could probably move this selector to the root of the stylesheet so it isn't nested in anything as 

```
.pf-c-data-list__toggle-icon {
    pointer-events: none;
    transition: var(--pf-c-data-list__toggle-icon--Transition);

    .pf-c-data-list__item.pf-m-expanded & {
        transform: var(--pf-c-data-list__item--m-expanded__toggle--c-button-icon--Transform);
   }
}
```